radis/lbl/base.py

```diff
@@ -3239,64 +3240,108 @@ def plot_populations(self, what="vib", isotope=None, nfig=None):
+        assert wavenum_min is not None and wavenum_max is not None
+        assert wavenum_min < wavenum_max
+        if not isinstance(wunit, Default):
+            raise ValueError("Please use wmin/wmax when passing wunit")
+        if isinstance(wavenum_min, u.Quantity) or isinstance(wavenum_max, u.Quantity):
+            assert isinstance(wavenum_min, u.Quantity) and isinstance(
+                wavenum_max, u.Quantity
+            )
+            assert wavenum_min.unit.is_equivalent(1 / u.cm)
+            assert wavenum_max.unit.is_equivalent(1 / u.cm)
+
+    if isinstance(wmin, u.Quantity) or isinstance(wmax, u.Quantity):
+        assert wmin is not None and wmax is not None
+        assert isinstance(wmin, u.Quantity) and isinstance(wmax, u.Quantity)
+        assert wmin.unit.is_equivalent(u.m) or wmin.unit.is_equivalent(1 / u.m)
+        assert wmax.unit.is_equivalent(u.m) or wmax.unit.is_equivalent(1 / u.m)
+        assert wmin.unit.is_equivalent(wmax.unit)
+        if not isinstance(wunit, Default):
+            if not wmin.unit.is_equivalent(u.Unit(wunit)):
+                raise ValueError("Conflicting units passed for wmin/wmax and wunit")
+            # Should I keep this in?
+            # Deals with cases like: calc_spectrum(wmin=10*u.cm, wmax=1.u.m, wunit="cm")
+    #             else:
+    #                 if wmin.unit != wmax.unit and (wmin.unit != u.Unit(wunit.value) or wmax.unit != u.Unit(wunit.value)):
+    #                     raise Warning("Ambiguous units passed, ignoring wunit")
+
+    if wmin is not None and wmax is not None:
+        if isinstance(wmin, u.Quantity) or isinstance(wmax, u.Quantity):
+            if wmin.unit.is_equivalent(u.m):
+                wavelength_min = wmin
+                wavelength_max = wmax
+            else:
+                wavenum_min = wmin
+                wavenum_max = wmax
+        else:
+            if isinstance(wunit, Default):
+                wavenum_min = wmin * u.Unit(wunit.value)
+                wavenum_max = wmax * u.Unit(wunit.value)
+            else:
+                if u.Unit(wunit).is_equivalent(u.m):
+                    wavelength_min = wmin * u.Unit(wunit)
+                    wavelength_max = wmax * u.Unit(wunit)
+                else:
+                    wavenum_min = wmin * u.Unit(wunit)
+                    wavenum_max = wmax * u.Unit(wunit)
+
+    if wavenum_min is not None or wavenum_max is not None:
+        wavenum_min = convert_and_strip_units(wavenum_min, 1 / u.cm)
+        wavenum_max = convert_and_strip_units(wavenum_max, 1 / u.cm)
+
+    if wavelength_min is not None or wavelength_max is not None:
+        assert medium in ["air", "vacuum"]
+        wavelength_min = convert_and_strip_units(wavelength_min, u.nm)
+        wavelength_max = convert_and_strip_units(wavelength_max, u.nm)
         if medium == "air":
             wavelength_min_vac = air2vacuum(wavelength_min)
             wavelength_max_vac = air2vacuum(wavelength_max)
```

Comment:
Or, if you find it nicer (no need to create this new _vac variables) :

```suggestion
        if medium == "air":
            wavenum_min = nm_air2cm(wavelength_max)
            wavenum_max = nm_air2cm(wavelength_min)
        else: # medium == 'vacuum':
            wavenum_min = nm2cm(wavelength_max)
            wavenum_max = nm2cm(wavelength_min)
```

Reply:
That definitely looks much cleaner! Thanks :D